### PR TITLE
Migrate from Gitlab CI to GitHub Actions

### PR DESCRIPTION
**Pull request by joren485**
### Summary
Replace the GitLab CI config with GitHub Actions config.

### CODE REVIEW

**Comment by Legosteen11** [comment]
```    - coverage report
  after_script:
    - coverage html --directory=covhtml --title="${CI_COMMIT_REF_SLUG} Coverage Report"
  artifacts:
    paths:
      - covhtml
```

Shall we handle these artifacts with this: https://github.com/actions/upload-artifact?

**Comment by Legosteen11** [comment]
Coverage report upload works!

**Comment by Legosteen11** [comment]
```
docs-tests:
  stage: test
  except:
    - schedules
  image: thalia/concrexit-dependencies
  before_script:
    - poetry install --no-interaction --extras "docs"
  script:
    - echo "Building current docs"
    - cd docs
    - env -u GITLAB_CI poetry run make doctest
    - env -u GITLAB_CI poetry run sphinx-build -W . _build
    - echo "Checking if there are changes"
    - poetry run ./generate-apidocs.sh
    - git diff --exit-code
  artifacts:
    paths:
      - docs/_build
```
This job in Gitlab CI seems a bit weird (e.g. why do we have these echos? Git diff?)

**Comment by Legosteen11** [comment]
Documentation generation also works.

Next we should think about better names that are easier to understand.

**Comment by joren485** [comment]
This is a test comment

**Comment by joren485** [comment]
deploy review

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+              working-directory: docs
+              run: poetry run make doctest
+
+            - name: Run Sphinx
+              working-directory: docs
+              run: poetry run sphinx-build -W . _build -j $(nproc)
+
+            - name: Run API docs script
+              working-directory: docs
+              run: poetry run ./generate-apidocs.sh
```
Comment (se-bastiaan) [inline .github/workflows/test.yaml]: This is missing the `git diff --exit-code`. There should be no changes for this job to succeed. Now there is no test, you're just generating the docs.

Comment (se-bastiaan) [reply .github/workflows/test.yaml]: I think that also answers the question from https://github.com/thaliawww/concrexit/pull/2#issuecomment-594867975

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+            - uses: rlespinasse/github-slug-action@master
+            - name: Sync files to S3
+              run: aws s3 sync --only-show-errors html "s3://thalia-coverage/${GITHUB_REF_SLUG}/"
+              env:
+                  AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+                  AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+            
+            - name: Show coverage report link
+              run: echo "You can view the coverage report on https://s3-eu-west-1.amazonaws.com/thalia-coverage/${GITHUB_REF_SLUG}/index.html"
```
Comment (se-bastiaan) [inline .github/workflows/test.yaml]: I know this is not really related to this issue, but the simpler url to reach a S3 bucket would be `https://thalia-documentation.s3.amazonaws.com/${GITHUB_REF_SLUG}/index.html`

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+
+            - uses: rlespinasse/github-slug-action@master 
+            - name: Sync files to S3
+              run: aws s3 sync --only-show-errors html "s3://thalia-documentation/${GITHUB_REF_SLUG}/"
+              env:
+                  AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+                  AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+
+            - name: Show documentation link
+              run: echo "You can view the documentation on https://s3-eu-west-1.amazonaws.com/thalia-coverage/${GITHUB_REF_SLUG}/index.html"
```
Comment (se-bastiaan) [inline .github/workflows/test.yaml]: This url is incorrect, it mentions coverage instead of documentation

**Pull request merged**